test/functional/wallet_listsinceblock.py

```diff
@@ -167,6 +169,31 @@ def test_reorg(self):
         found = next(tx for tx in transactions if tx['txid'] == senttx)
         assert_equal(found['blockheight'], self.nodes[0].getblockheader(nodes2_first_blockhash)['height'])
 
+    def test_cant_read_block(self):
+        self.log.info('Test passing cannot read block from disk rpc error')
+
+        # Split network into two
+        self.split_network()
+
+        # send to nodes[0] from nodes[2]
+        self.nodes[2].sendtoaddress(self.nodes[0].getnewaddress(), 1)
+
+        # generate on both sides
+        nodes1_last_blockhash = self.generate(self.nodes[1], 6, sync_fun=lambda: self.sync_all(self.nodes[:2]))[-1]
+        self.generate(self.nodes[2], 7, sync_fun=lambda: self.sync_all(self.nodes[2:]))[0]
+
+        self.join_network()
+
+        # Setting permissions to none
+        os.rename(self.nodes[0].blocks_path / "blk00000.dat", self.nodes[0].blocks_path / "blk00000.dat.copy")
```

Comment:
nit:

This `self.nodes[0].blocks_path / "blk..." / ...` is a [`Path`](https://docs.python.org/3/library/pathlib.html) so you don't need to `import os` to call `os.rename` to rename it, you can instead:

```suggestion
        (self.nodes[0].blocks_path / "blk00000.dat").rename("blk00000.dat.copy")
```

Most of the things you could do to files/directories with `os` can also be done with `Path`.

(The `.rename` is aware of which bit of the path is the file so you don't need to give it the `self.nodes[0].blocks_path` for the rename to work correctly)

Reply:
thanks! added in [21bc4ff](https://github.com/bitcoin/bitcoin/pull/30195/commits/21bc4ff147cdd0a2fd37097feffbab8edfc93320)

One thing I did notice is that if I did not add the full path in `.rename` the second time I call `.rename` it cannot find the correct file and when I check the blocks dir manually it seemed like it never changed the file name. But oddly the rpc error is still seen

I might do some more investigation why I'm seeing that, but let me know if this looks okay now

Reply:
Hrm, I can't check just now but I'd guess either `.rename` returns a new path instance that you should use for further changes instead, or you can try call `.resolve()` on it after rename? (Resolves to an absolute path, maybe needed if the original changed?)

Reply:
https://docs.python.org/3/library/pathlib.html#pathlib.Path.rename

> The target path may be absolute or relative. Relative paths are interpreted relative to the current working directory, not the directory of the Path object.

Maybe this?

Reply:
I've added https://github.com/bitcoin/bitcoin/pull/30195#discussion_r1630897082 with a suggestion